services/sso/src/test/java/software/amazon/awssdk/services/sso/internal/SsoAccessTokenProviderTest.java

```diff
@@ -127,6 +131,44 @@ public void cachedTokenFile_tokenFileNotExist_throwNullPointerException() throws
         assertThatThrownBy(() -> provider.resolveToken().token()).isInstanceOf(UncheckedIOException.class);
     }
 
+    @Test
+    void cachedTokenFile_AboutToExpire_resolveAccessTokenCorrectly() throws IOException {
+        String tokenFile = String.format("{\n" +
+                                         "\"accessToken\": \"base64string\",\n" +
+                                         "\"expiresAt\": \"%s\",\n" +
+                                         "\"startUrl\": \""+ START_URL +"\"\n" +
+                                         "}", stringFormattedTime(Instant.now().plusSeconds(10)));
+        SsoAccessTokenProvider provider = new SsoAccessTokenProvider(
+            prepareTestCachedTokenFile(tokenFile, GENERATED_TOKEN_FILE_NAME));
+        assertThat(provider.resolveToken().token()).isEqualTo("base64string");
+    }
+
+    @Test
+    void cachedTokenFile_JustExpired_resolveAccessTokenCorrectly() throws IOException {
```

Comment:
nit: it seems test name needs to be updated since we are testing exceptional case.

Reply:
done